kyo-core/shared/src/main/scala/kyo/loops.scala

```diff
@@ -107,6 +107,27 @@ object Loops:
         loop(input1, input2, input3)
     end transform
 
+    inline def indexed[Output: Flat, S](
+        inline run: Int => Result[Unit, Output] < S
+    ): Output < S =
+        def _loop(idx: Int): Output < S =
+            loop(idx)
```

Comment:
Why is this necessary?

Reply:
On each iteration of a loop execution, it can either return a pure value, which triggers immediate recursion with `@tailrec`, or a suspended computation, which then suspends the loop via `map`. When the loop is suspended, it needs to start again once the computation resumes but it isn't a tail-recursive call, and it doesn't need to be since the effect suspension will unfold the stack. This method alias allows recursion in such cases while still keeping the `@tailrec` annotation for the fast path when there's no suspension.

Reply:
Ah, thanks for the concise explanation. Very clever.